src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -2982,19 +2977,17 @@ static int icap_cache_bitstream_axlf_section(struct platform_device *pdev,
 
 	if (kind == MEM_TOPOLOGY || kind == ASK_GROUP_TOPOLOGY) {
 		struct mem_topology *mem_topo = *target;
-		u64 hbase, hsz;
 		int i;
 
 		for (i = 0; i< mem_topo->m_count; ++i) {
 			if (!IS_HOST_MEM(mem_topo->m_mem_data[i].m_tag) ||
 			    mem_topo->m_mem_data[i].m_used)
 				continue;
 
-			if (!xocl_m2m_host_bank(xdev, &hbase, &hsz)) {
-				mem_topo->m_mem_data[i].m_used = 1;
-				mem_topo->m_mem_data[i].m_base_address = hbase;
-				mem_topo->m_mem_data[i].m_size = (hsz >> 10);
-			}
+			xocl_m2m_host_bank(xdev,
+				&(mem_topo->m_mem_data[i].m_base_address),
+				&(mem_topo->m_mem_data[i].m_size),
+				&(mem_topo->m_mem_data[i].m_used));
 		}
```

Comment:
I don't recall I added this piece of code, please have @houlz0507 review this login if it is correct.

Reply:
this part need to change since the host bank info may come from shell (u50 nodma case) or the xclbin itself (there is no shell metadata at all)
this pr moves the 	icap_cache_bitstream_axlf_section(pdev, xclbin, ASK_GROUP_TOPOLOGY) to after the m2m subdevice is created.
please think about it whether it has issue.

Reply:
This sequence should be fine during normal driver attach. But I recall the icap won't be teared down (remove/probe) during reset, instead the icap has offline/online callbacks. Please double check if "reset" also works as expected.

Reply:
this change is only related to xclbin download. I don't think it will impact reset.
xclbin needs to be reloaded anyway after reset